CalibParam:  
  # 内参校准参数（已更新）
  INTRI:
    Camera:
      - key: /image_topic
        value:
          polymorphic_id: 2147483649
          polymorphic_name: pinhole_brown_t2
          ptr_wrapper:
            id: 2147483650
            data:
              img_width: 1280
              img_height: 1024
              # focal_length:                     # 8mm相机
              #   - 1740.1391026563904
              #   - 1743.466162735272
              # principal_point:
              #   - 600.0573645949644
              #   - 415.5957223704912
              # disto_param:
              #   - -0.11407724406859104
              #   - 0.9907393620859306
              #   - -0.012350441691573684
              #   - -0.0022510643311303606
              #   - -2.920015848738005
              focal_length:                     # 6mm相机
                - 1330.649408921195
                - 1332.3006999890033
              principal_point:
                - 627.2791611048931
                - 533.2806702069377
              disto_param:
                - -0.14828277112729327
                - 0.7868245788660153
                - 0.0023478362007317123
                - 0.002535817596959765
                - -1.5222934363748177
              # focal_length:                   # 4mm相机
              #   - 893.4306170676092
              #   - 888.878979439582
              # principal_point:
              #   - 618.4622245880115
              #   - 511.73226091888046
              # disto_param:
              #   - -0.09613598816721357
              #   - 0.3686128765511766
              #   - -0.0035685183642176046
              #   - 0.002876545526120485
              #   - -0.594006385480322
  


# 相机参数
camera:
  parameters:
    exposure_ms: 5.0 
    gain: 12.0 
    fps: 120.0
    target_width: 1280
    target_height: 1024
    image_topic: "image_topic"
    debug: false

# 神经网络识别参数
yolo11:
  yolo11_model_path: "/home/guo/ITL_Auto_aim/src/model/yolo11.xml"
  # 推理设备
  device: "CPU"  # 可选: CPU, GPU, MYRIAD
  enemy_color: "blue"
  # 检测阈值
  threshold: 0.5
  min_confidence: 0.8
  score_threshold: 0.7
  nms_threshold: 0.3


# 传统方法识别参数：
classify_model: "/home/guo/ITL_Auto_aim/src/model/tiny_resnet.onnx"
threshold: 150
max_angle_error: 45 # degree
min_lightbar_ratio: 1.5
max_lightbar_ratio: 20
min_lightbar_length: 8
min_armor_ratio: 1
max_armor_ratio: 5
max_side_ratio: 1.5
max_rectangular_error: 25 # degree
min_confidence: 0.8


# DM_IMU相关配置
DM_IMU:
# 串口配置
imu_com_port: "/dev/ttyACM1"
baud: 921600

# 发布频率相关
publish_rate: 1000  # Hz
  

# 坐标系配置
# 相机坐标系即标准的相机坐标系，以相机为视角向前看，向前是Z轴正向，向右是X轴正向，向下是Y轴正向
# 变换之后的云台坐标系，向前是X轴正向，向上是Z轴正向，向左是Y轴正向


# 解算模块所需的参数
Solver:
  coord_converter:  # 坐标变换所需参数
    rotation_matrix_camera_to_gimbal:
      rows: 3
      cols: 3
      dt: float
      data: [0, 0, 1,
            -1, 0, 0,
            0, -1, 0]

    rotation_matrix_gimbal_to_imu:
      rows: 3
      cols: 3
      dt: float
      data: [-1, 0, 0,
              0, -1, 0,
              0, 0, 1]

# 陀螺仪与相机放置在同一位置
# - x: 相机相对云台在前后方向的偏移
# - y: 相机相对云台在左右方向的偏移
# - z: 相机相对云台在上下方向的偏移

t_camera_to_gimbal:
[0.00, 0.00, 0.08]
# [0.115, -0.065, 0.015]


# Tracker参数
Tracker:
  enemy_color: "red"  # 或 "blue"
  # 跟踪状态机参数
  min_detect_count: 3              # 最小检测次数
  max_temp_lost_count: 10          # 普通目标的最大临时丢失容忍度  
  outpost_max_temp_lost_count: 20  # 前哨站的最大临时丢失容忍度

enemy_color: "red"  # 或 "blue"
# 跟踪状态机参数
min_detect_count: 3              # 最小检测次数
max_temp_lost_count: 10          # 普通目标的最大临时丢失容忍度  
outpost_max_temp_lost_count: 20  # 前哨站的最大临时丢失容忍度

# 瞄准器Aimer参数
Aimer:
  #####-----aimer参数-----#####
  yaw_offset: -0.8 # degree -2.5
  pitch_offset: -1  # degree 2
  comming_angle: 60 # degree
  leaving_angle: 20 # degree
  left_yaw_offset: -1
  right_yaw_offset: -0.6
  decision_speed: 10 # rad/s
  high_speed_delay_time: 0.026 # s
  low_speed_delay_time: 0.010 # s


###planner参数###
fire_thresh: 0.0035

max_yaw_acc: 50
Q_yaw: [9e6, 0]
R_yaw: [1]

max_pitch_acc: 100
Q_pitch: [9e6, 0]
R_pitch: [1]
yaw_offset: 2 # degree -2.5
pitch_offset: 6.5 # degree 2
comming_angle: 55 # degree
leaving_angle: 20 # degree
decision_speed: 7 # rad/s
high_speed_delay_time: 0.0 # s
low_speed_delay_time: 0.0 # s planner use this value

# Gimbal通讯参数
Gimbal:
enable_serial: true
com_port: "/dev/ttyACM0"
com_port_1: "/dev/ttyACM2"


# video视频测试参数
Video:
video_path: "/home/guo/ITL_Auto_aim/test/31313561759-1-192.mp4"

#####-----buff_detector参数-----#####
buff_model: "/home/guo/ITL_Auto_aim/src/model/yolo11_buff_int8.xml"

#####-----buff_aimer参数-----#####
fire_gap_time: 0.700   # s
predict_time: 0.120     # s
  
# 基地Base识别参数
Base_Hit:
  Openvino_XML: /home/guo/ITL_Auto_aim/src/model/Katrin.xml
  Openvino_Deveice: "CPU"
  input_width: 640
  input_height: 384
  score_threshold: 0.6
  nms_threshold: 0.3